src/EFCore.PG/Query/Internal/NonQueryConvertingExpressionVisitor.cs

```diff
@@ -0,0 +1,82 @@
+using Npgsql.EntityFrameworkCore.PostgreSQL.Query.Expressions.Internal;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.Internal;
+
+/// <summary>
+///     Converts the relational <see cref="NonQueryExpression" /> into a PG-specific <see cref="PostgresDeleteExpression" />, which
+///     precisely models a DELETE statement in PostgreSQL. This is done to handle the PG-specific USING syntax for table joining.
+/// </summary>
+public class NonQueryConvertingExpressionVisitor : ExpressionVisitor
```

Comment:
@smitpatel here's the conversion to the PG-specific delete expression, as discussed in https://github.com/npgsql/efcore.pg/pull/2449#discussion_r931532137.